R/fitNullModel.R

```diff
@@ -71,9 +82,10 @@ setMethod("fitNullModel",
                   ind <- match(.covMatNames(cov.mat), rownames(x))
                   x <- x[ind,]
               }
-              
+
               nm <- fitNullModel(x, outcome, covars, cov.mat, group.var, ...)
-              nm$sample.id <- rownames(nm$model.matrix)
+              # XXX: Decide if we should reorder columns in the fit data frame such that sample.id is the first column.
+              nm$fit$sample.id <- rownames(nm$model.matrix)
```

Comment:
When we use an AnnotatedDataFrame, are we going to end up with the sample.id being both a column and the rownames of the `fit` object?

Reply:
Right now, yes. I could remove the rownames when adding the `sample.id` column without breaking anything, I think.

Reply:
Also noting if we removed the rownames when `sample.id` was present, we'd have to do it in a few wrappers (or write an internal function to add `sample.id` to the null model fit data frame, and then remove the rownames).

Reply:
It probably doesn't really matter. If we want it to be "cleaner", the last thing you said seems reasonable. We could just have a function that operates at the end of fitting the null model that adds a `sample.id` column and removes the rownames if needed. Then the `fit` matrix would always look the same regardless of input data type, which might be nice. Your call.

Reply:
I think @smgogarten said on Slack that the differences with `sample.id` were intentional for different input types (AnnotatedDataFrame vs. data.frame), so I tried to it keep it that way. I'm leaning towards just keeping it as is.